crates/turborepo-paths/src/absolute_system_path.rs

```diff
@@ -221,11 +221,54 @@ impl AbsoluteSystemPath {
     pub fn components(&self) -> Utf8Components<'_> {
         self.0.components()
     }
+
+    pub fn collapse(&self) -> AbsoluteSystemPathBuf {
```

Comment:
Do we actually need this? I thought we were keeping these paths "clean" already...

Reply:
Not everywhere apparently. And specifically with constructing an `AbsoluteSystemPath`, you can't clean on construction because that might require allocation

Reply:
But I don't know whether we need this specifically versus just using path-clean

Reply:
> Not everywhere apparently.
That sounds like a bug. Can you file a ticket for it?